### PR TITLE
RCD windoor fixes

### LINKED ISSUE #64742: Windoors made from RCDs issues
Reporting client version: 514.1571

## Round ID:
[177994](https://scrubby.melonmesa.com/round/177994)

## Testmerges:
- [Ballistic to Energy: Autorifles for Thermal Pistols](https://www.github.com/tgstation/tgstation/pull/64280)
- [Refactor: make SSrunechat to be subsystem of SStimer](https://www.github.com/tgstation/tgstation/pull/64366)

## Reproduction:

1. Windoors don't receive the name set in the access settings and just have the default one
2. Unrestricted access west worked on a west-facing windoor but not an east-facing one
3. The one access option doesn't work, which prevents windoors being opened at all (not sure if it works if you have all the accesses)

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

- Makes require one access work properly with windoors created by the RCD
- Makes unrestricted directional access for windoors work properly when opening them from the same tile
- Makes naming airlocks with the RCD work for windoors
- Adds unrestricted access overlays to windoors

Space Station 13 2022-03-03 221030

Closes https://github.com/tgstation/tgstation/issues/64742

## Why It's Good For The Game

Fixes bugs and makes unrestricted windoor access obvious.

## Changelog

:cl:
fix: Windoors created by the RCD are correctly named
fix: Windoors created by the RCD work properly with require one access
fix: Windoors with unrestricted directional access open properly when operated from the same tile
qol: Windoors have unrestricted directional access overlays like airlocks now
/:cl:

### CODE REVIEW

File: code/game/machinery/doors/door.dm [hunk 1]
```diff
@@ -313,8 +316,51 @@
...
+		set_light(0)
+
 	return ..()
 
+/obj/machinery/door/update_overlays()
+	. = ..()
+
+	if(hasPower() && unres_sides)
+		if(unres_sides & NORTH)
+			var/image/I = image(icon='icons/obj/doors/airlocks/station/overlays.dmi', icon_state="unres_n")
```
Comment (SuperNovaa41) [inline code/game/machinery/doors/door.dm:332]: da funny bad var names (change this NOW)

File: code/game/machinery/doors/door.dm [hunk 2]
```diff
@@ -313,8 +316,51 @@
...
+		set_light(2, 1)
+	else
+		set_light(0)
+
 	return ..()
 
+/obj/machinery/door/update_overlays()
+	. = ..()
+
+	if(hasPower() && unres_sides)
```
Comment (SuperNovaa41) [inline code/game/machinery/doors/door.dm:330]: This should be an early return

File: code/game/machinery/doors/door.dm [hunk 3]
```diff
@@ -313,8 +316,51 @@
...
+					I.pixel_x = 6
+			. += I
+		if(unres_sides & WEST)
+			var/image/I = image(icon='icons/obj/doors/airlocks/station/overlays.dmi', icon_state="unres_w")
+			switch(dir)
+				if(EAST)
+					I.pixel_x = -6
+				if(WEST)
+					I.pixel_x = -31
+			. += I
```
Comment (SuperNovaa41) [inline code/game/machinery/doors/door.dm:362]: I feel like this whole thing could probably be condensed down to two if statements

Comment (cacogen) [reply code/game/machinery/doors/door.dm:362]: Well let me know if you figure out how.............

File: code/game/machinery/doors/windowdoor.dm [hunk 4]
```diff
@@ -40,6 +40,20 @@
 	if(cable)
 		debris += new /obj/item/stack/cable_coil(src, cable)
 
+	src.unres_sides = unres_sides
+
+	if(src.unres_sides)
```
Comment (Mothblocks) [inline code/game/machinery/doors/windowdoor.dm:45]: Do all this before setting `src.unres_sides` so none of this needs to use `src`

File: code/game/machinery/doors/windowdoor.dm [hunk 5]
```diff
@@ -67,6 +81,51 @@
 	. = ..()
 	icon_state = "[base_state][density ? null : "open"]"
 
+	if(hasPower() && unres_sides)
+		set_light(2, 1)
```
Comment (Mothblocks) [inline code/game/machinery/doors/windowdoor.dm:85]: Named parameters, if the linter doesn't get mad about it.

File: code/game/machinery/doors/windowdoor.dm [hunk 6]
```diff
@@ -67,6 +81,51 @@
...
+		return
+
+	set_light(0)
```
Comment (Mothblocks) [inline code/game/machinery/doors/windowdoor.dm:88]: Named parameters, if the linter doesn't get mad about it.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by cacogen** [comment]
@Mothblocks I think this is pretty close to perfect now if you want to review again when you've got time

**Pull request merged**